packages/daemon/src/daemon-node-powers.js

```diff
@@ -310,6 +310,20 @@ export const makePowers = ({
     return fs.promises.readFile(path, 'utf-8');
   };
 
+  /**
+   * @param {string} path
+   */
+  const maybeReadFileText = async path =>
+    readFileText(path).catch(error => {
+      if (
+        error.message.startsWith('ENOENT: ') ||
+        error.message.startsWith('EISDIR: ')
+      ) {
+        return undefined;
+      }
+      throw error;
+    });
```

Comment:
This has been a useful pattern in other tools. Testing existence with `stat` is an anti-pattern, so I favor attempting to read and handling failure if the file didn’t exist, passing the data forward if the test succeeds. But, classifying the errors is necessary to avoid swallowing file descriptor limit errors.